### PR TITLE
Cleanup headers of libYARP_robotinterface

### PR DESCRIPTION
**Pull request by traversaro**
* Fix include guards  of RobotInterfaceDTD
* Make XMLReaderFileVx, XMLReaderFileV1 and XMLReaderFileV3 private classes

### CODE REVIEW

File: src/libYARP_robotinterface/src/yarp/robotinterface/impl/RobotInterfaceDTD.h [hunk 1]
```diff
@@ -6,6 +6,8 @@
  * BSD-3-Clause license. See the accompanying LICENSE file for details.
  */
 
+#ifndef YARP_ROBOTINTERFACE_ROBOTINTERFACEDTD_H
+#define YARP_ROBOTINTERFACE_ROBOTINTERFACEDTD_H
```
Comment (drdanz) [inline src/libYARP_robotinterface/src/yarp/robotinterface/impl/RobotInterfaceDTD.h]: Nitpick: `YARP_ROBOTINTERFACE_IMPL_ROBOTINTERFACEDTD_H`

File: src/libYARP_robotinterface/src/yarp/robotinterface/impl/XMLReaderFileVx.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_ROBOTINTERFACE_XMLREADERFILEVX_H
+#define YARP_ROBOTINTERFACE_XMLREADERFILEVX_H
```
Comment (drdanz) [inline src/libYARP_robotinterface/src/yarp/robotinterface/impl/XMLReaderFileVx.h]: `YARP_ROBOTINTERFACE_IMPL_XMLREADERFILEVX_H`

**Comment by traversaro** [comment]
Done, do you want me to squash manually or it is ok to merge with squash?

**Comment by drdanz** [comment]
Let's see if this works: ba28c14e10fbb768f76f89aba853ce73f0263607

/squash

**Comment by drdanz** [comment]
@traversaro Apparently the push on forks is no longer working, I tried also doing it manually, but it fails on the push... Do you mind rebasing and squashing the commits manually?

**Comment by traversaro** [comment]
Ok no problem!

**Comment by traversaro** [comment]
> Ok no problem!

Done.

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**